code/__HELPERS/unsorted.dm

```diff
@@ -1370,3 +1370,21 @@ var/mob/dview/dview_mob = new
 // call to generate a stack trace and print to runtime logs
 /proc/crash_with(msg)
 	CRASH(msg)
+
+/atom/proc/find_up_hierarchy(var/atom/target)

```

Comment:
This proc doesn't appear to be recursive? A recursive proc would reference itself. Right now, it appears to only run one actual cycle 10 times?